src/ast.mli

```diff
@@ -0,0 +1,91 @@
+type program = None
+
+type bin_op =
+  Add    | Minus | Multiply | Divide
+  | Modulo | Equal | Neq      | Lt
+  | Leq    | Gt    | Geq      | And
+  | Or
+
+type u_op = Not | Neg
+
+type p_type = Int_t | Bool_t | Double_t | Char_t | Unit_t
+
+type cont_type = String_t | List_t | Set_t | Map_t | Tuple_t
+
+type actor_op =
+  | Actor_send
+  | Actor_broadcast
+  | Actor_receive
+
+type types =
```

Comment:
Quite possible, I wasn't sure about my implementation of opt. 

One thing that's possible with this definition is:
`None<int> = ...` ,
which should not be allowed.